src/packagedcode/rpm.py

```diff
@@ -319,36 +319,4 @@ def detect_using_name_mapping(declared, licensing=Licensing()):
     """
     # TODO: use RPM symbology
     declared = declared.lower()
-    detected = get_declared_to_detected().get(declared)
-    if detected:
-        return str(licensing.parse(detected, simple=True))
-
-
-_DECLARED_TO_DETECTED = None
-
-
-def get_declared_to_detected(data_file=None):
-    """
-    Return a mapping of declared to detected license expression cached and
-    loaded from a tab-separated text file, all lowercase, normalized for spaces.
-
-    This data file is from license keys used in RPMs files and should be
-    derived from a large collection of RPMs files.
-    """
-    global _DECLARED_TO_DETECTED
-    if _DECLARED_TO_DETECTED is not None:
-        return _DECLARED_TO_DETECTED
-
-    _DECLARED_TO_DETECTED = {}
-    if not data_file:
-        data_file = os.path.join(os.path.dirname(__file__), 'rpm_licenses.txt')
-    with open(data_file) as df:
-        for line in df:
-            line = line.strip()
-            if not line or line.startswith('#'):
-                continue
-            decl, _, detect = line.partition('\t')
-            if detect and detect.strip():
-                decl = decl.strip()
-                _DECLARED_TO_DETECTED[decl] = detect
-    return _DECLARED_TO_DETECTED
+    return str(licensing.parse(declared, simple=True))
```

Comment:
In removing all this, are you sure you are solving the issue?

Reply:
Ok so in above code we are getting list of rpm licenses from this [file](https://github.com/nexB/scancode-toolkit/blob/develop/src/packagedcode/rpm_licenses.txt) but file is not populated yet so this function is returning None and because detected is none it is not parsing already detected license expression causing license-expression to be unknown

Reply:
,
      "keywords": [],
      "homepage_url": "http://projects.gnome.org/gconf/",
      "download_url": null,
      "size": null,
      "sha1": null,
      "md5": null,
      "sha256": null,
      "sha512": null,
      "bug_tracking_url": null,
      "code_view_url": null,
      "vcs_url": null,
      "copyright": null,
      "license_expression": "lgplv2+ AND gplv2+",
      "declared_license": "LGPLv2+ and GPLv2+",
      "notice_text": null,
      "contains_source_code": null,
      "source_packages": [
        "pkg:rpm/GConf2@3.2.6-31.fc36?arch=src"
      ],
      "extra_data": {},
      "package_uuid": "pkg:rpm/GConf2-devel@3.2.6-31.fc36?uuid=52f31f49-9685-4d0d-887e-df30afcd74ae",
      "package_data_files": [
        "GConf2-devel-3.2.6-31.fc36.aarch64.rpm"
      ],

Reply:
As an example, in this:
- "declared_license": "LGPLv2+ and GPLv2+",

we should obtain this
- "license_expression": "lgpl-2.0-plus AND gpl-2.0-plus",

and not this:
- "license_expression": "lgplv2+ AND gplv2+",


So IMHO you need to build a list of "symbols" used in RPM licensing such that we can then parse the expression correctly?

Reply:
ah yes I also thought of that can you tell me how to specify those symbols in [rpm_license.txt](https://github.com/nexB/scancode-toolkit/blob/develop/src/packagedcode/rpm_licenses.txt) file

Reply:
> ah yes I also thought of that can you tell me how to specify those symbols in [rpm_license.txt](https://github.com/nexB/scancode-toolkit/blob/develop/src/packagedcode/rpm_licenses.txt) file

I would use either a simple tab separator (two columns: rpm_license, scancode_expression, or you can make this a JSON or YAML file instead

Reply:
gentle ping... do not hesitate to ask questions if you are not sure how to proceed!

Reply:
I've worked on this and found a solution I'll update the PR